CppSamples/DisplayInformation/ShowGrid/ShowGrid.qml

```diff
@@ -19,24 +24,116 @@ import QtQuick.Controls
 import QtQuick.Layouts
 import Esri.Samples
 
-DisplayGridSample {
-    id: displayGrid
-    clip: true
-    width: 800
-    height: 600
-
-    property real centerWindowY: (displayGrid.height / 2) - (styleWindow.height / 2)
+Item {
+    id: root
+    property real centerWindowY: (root.height / 2) - (styleWindow.height / 2)
 
     MapView {
         id: mapQuickView
         anchors.fill: parent
-
+        visible: gridSample.currentViewType == "MapView"
         Component.onCompleted: {
-            // Set the focus on MapView to initially enable keyboard navigation
+            // Set and keep the focus on SceneView to enable keyboard navigation
             forceActiveFocus();
         }
     }
-    mapView: mapQuickView
+
+    SceneView {
+        id: sceneQuickView
+        anchors.fill: parent
+        visible: gridSample.currentViewType == "SceneView"
+    }
+
+    // Declare the C++ instance which creates the map, scene etc. and supply the views
+    ShowGridSample {
+        id: gridSample
+        mapView: mapQuickView
+        sceneView: sceneQuickView
+
+        onCurrentViewTypeChanged: {
+            if (gridSample.currentViewType === "MapView")
+                mapQuickView.forceActiveFocus();
+            else
+                sceneQuickView.forceActiveFocus();
+        }
+    }
+
+    Row {
+        id: viewButtonsRow
+        anchors
+        {
+            top: parent.top
+            horizontalCenter: parent.horizontalCenter
+            topMargin: 5
+        }
+
+        spacing: 2
+
+        Rectangle {
+            id: mapViewButton
+            property bool selected: gridSample.currentViewType === "MapView"
```

Comment:
☁️ all of these strings floating around is what makes me think an enum would be better

Reply:
That's fair, but we only do two string comparisons on the front end and setting up enums for all (or some) of the properties might get a lot more cumbersome